ChangeLog.md

```diff
@@ -44,6 +44,7 @@ See docs/process.md for more on how version tagging works.
 - A new settings `TABLE_BASE` was introduced that can be used to place static
   function addresses (table slots) at a certain offset.  This defaults to 1
   which is the previously fixed value. (#20149)
+- Clang's error detection of unused variables improved. As a result, three unused variables were removed. (#20169)
```

Comment:
```suggestion
- Clang's error detection of unused variables improved, which may require changes in user code (see #20169).
```

Reply:
(just to refocus on the impact to users, who are the main people reading this, I think)

Reply:
+1, the important thing is that the change may affect the warnings users see. You could also mention that it's the `-Wunused-variable` warning that got smarter. That way, users would know that they could use `-Wno-unused-variable` to disable the warning if they can't modify the code. (This is also why clang prints the warning flag name along with the warning itself)